src/code42_connector.py

```diff
@@ -0,0 +1,238 @@
+# Python 3 Compatibility imports
+from __future__ import print_function, unicode_literals
+
+import json
+
+import py42.sdk
+from py42.services.detectionlists.departing_employee import DepartingEmployeeFilters
+from py42.services.detectionlists.high_risk_employee import HighRiskEmployeeFilters
+import requests
+
+# Phantom App imports
+import phantom.app as phantom
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+
+
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class Code42Connector(BaseConnector):
+    TEST_CONNECTIVITY_ACTION_ID = "test_connectivity"
+    ADD_DEPARTING_EMPLOYEE_ACTION_ID = "add_departing_employee"
+    REMOVE_DEPARTING_EMPLOYEE_ACTION_ID = "remove_departing_employee"
+    LIST_DEPARTING_EMPLOYEES_ACTION_ID = "list_departing_employees"
+    ADD_HIGH_RISK_EMPLOYEE_ACTION_ID = "add_highrisk_employee"
+    REMOVE_HIGH_RISK_EMPLOYEE_ACTION_ID = "remove_highrisk_employee"
+    LIST_HIGH_RISK_EMPLOYEES_ACTION_ID = "list_highrisk_employees"
+
+    def __init__(self):
+        super(Code42Connector, self).__init__()
+
+        self._state = None
+        self._cloud_instance = None
+        self._username = None
+        self._password = None
+        self._client = None
+        self._action_map = {
+            self.TEST_CONNECTIVITY_ACTION_ID: self._handle_test_connectivity,
+            self.ADD_DEPARTING_EMPLOYEE_ACTION_ID: self._handle_add_departing_employee,
+            self.REMOVE_DEPARTING_EMPLOYEE_ACTION_ID: self._handle_remove_departing_employee,
+            self.LIST_DEPARTING_EMPLOYEES_ACTION_ID: self._handle_list_departing_employees,
+            self.ADD_HIGH_RISK_EMPLOYEE_ACTION_ID: self._handle_add_high_risk_employee,
+            self.REMOVE_HIGH_RISK_EMPLOYEE_ACTION_ID: self._handle_remove_high_risk_employee,
+            self.LIST_HIGH_RISK_EMPLOYEES_ACTION_ID: self._handle_list_high_risk_employees
+        }
+
+    def initialize(self):
+        # use this to store data that needs to be accessed across actions
+        self._state = self.load_state()
+
+        config = self.get_config()
+        self._cloud_instance = config["cloud_instance"]
+        self._username = config["username"]
+        self._password = config["password"]
+
+        return phantom.APP_SUCCESS
+
+    def handle_action(self, param):
```

Comment:
I made `handle_action` take care of:
- creating the client
- creating and adding the action result (which, in addition to `param` is passed to all actions now)
- logging which action is about to be handled

This removed a fair amount of boilerplate code from the actions.